keyserver/listener.go

```diff
@@ -10,11 +10,16 @@ import (
 	. "github.com/coniks-sys/coniks-go/protocol"
 )
 
-func (server *ConiksServer) listenForRequests(ln *net.TCPListener, handler func(msg []byte) ([]byte, error)) {
+func (server *ConiksServer) listenForRequests(ln net.Listener, handler func(msg []byte) ([]byte, error)) {
```

Comment:
Should we have separate `listenForRequests` for `net.TCPListener` (which is the public connection) and `net.UnixListener` (which is the unix socket connection), to eliminate type switches below? This way, maybe we will have duplicate code but a better performance? 
/cc @arlolra @masomel @Liamsi